app/upgrades/v0.0.12.go

```diff
@@ -0,0 +1,154 @@
+		applyNewParameters := func(ctx context.Context) (err error) {
+			logger := cosmosTypes.UnwrapSDKContext(ctx).Logger()
+			logger.Info("Starting parameter updates", "upgrade_plan_name", Upgrade_0_0_12_PlanName)
+
+			// Set supplier module staking_fee to 1000000upokt, in line with the config.yml in the repo.
+			// Verify via:
+			// $ poktrolld q supplier params --node=...
+			supplierParams := keepers.SupplierKeeper.GetParams(ctx)
+			supplierParams.MinStake = &cosmosTypes.Coin{
+				Denom:  "upokt",
+				Amount: math.NewInt(supplierStakingFee),
+			}
+			err = keepers.SupplierKeeper.SetParams(ctx, supplierParams)
+			if err != nil {
+				logger.Error("Failed to set supplier params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated supplier params", "new_params", supplierParams)
+
+			// Add service module `target_num_relays` parameter, in line with the config.yml in the repo.
+			// Verify via:
+			// $ poktrolld q service params --node=...
+			serviceParams := keepers.ServiceKeeper.GetParams(ctx)
+			serviceParams.TargetNumRelays = serviceTargetNumRelays
+			err = keepers.ServiceKeeper.SetParams(ctx, serviceParams)
+			if err != nil {
+				logger.Error("Failed to set service params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated service params", "new_params", serviceParams)
+
+			// Add tokenomics module `global_inflation_per_claim` parameter, in line with the config.yml in the repo.
+			// Verify via:
+			// $ poktrolld q tokenomics params --node=...
+			tokenomicsParams := keepers.TokenomicsKeeper.GetParams(ctx)
+			tokenomicsParams.GlobalInflationPerClaim = tokenomicsGlobalInflationPerClaim
+			err = keepers.TokenomicsKeeper.SetParams(ctx, tokenomicsParams)
+			if err != nil {
+				logger.Error("Failed to set tokenomics params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated tokenomics params", "new_params", tokenomicsParams)
+			return nil
+		}
+
+		// Helper function to update all suppliers' RevShare to 100%. This is necessary to ensure that
+		// we have that value populated before suppliers are connected.
+		updateSuppliersRevShare := func(ctx context.Context) error {
+			logger := cosmosTypes.UnwrapSDKContext(ctx).Logger()
+			suppliers := keepers.SupplierKeeper.GetAllSuppliers(ctx)
+			logger.Info("Updating all suppliers to have a 100% revshare to the supplier",
+				"num_suppliers", len(suppliers))
+
+			for _, supplier := range suppliers {
+				for _, service := range supplier.Services {
+					// Add warning log if we're overwriting existing revshare settings. We can't get the previous
+					// revshare percentage due to a protobuf change, but we can at least log the number of previous revshares.
+					if len(service.RevShare) > 1 {
+						logger.Warn(
+							"Overwriting existing revenue share configuration",
+							"supplier", supplier.OperatorAddress,
+							"service", service.ServiceId,
+							"previous_revshare_count", len(service.RevShare),
```

Comment:
Can we logs the entire previous rev share, not just the count?

Reply:
Added a list of addresses.